pkg/api/security/security.go

```diff
@@ -256,3 +256,19 @@ func saveAuthToken(token, tokenPath string) error {
 	log.Infof("Wrote auth token in %s", tokenPath)
 	return nil
 }
+
+// call the file.Sync method to flush the file to disk and catch potential I/O errors
+func writeFileAndSync(name string, data []byte, perm os.FileMode) error {
+	f, err := os.OpenFile(name, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, perm)
+	if err != nil {
+		return err
+	}
+	_, err = f.Write(data)
+	if err1 := f.Sync(); err1 != nil && err == nil {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Why trying to `f.Sync()` before checking possible error returned by `f.Write(data)` ?

Reply:
I initially wanted to always have the same Write/Sync/Close sequence regardless of errors, but I agree that trying to sync data when the write failed makes little sense.
I changed the behavior so that the sync is only performed if the write operation was successful (96183e04c66846e894176ec8eeebb17e0fbec514)